apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.UploadObject.cs

```diff
@@ -48,6 +48,19 @@ public sealed partial class StorageClientImpl : StorageClient
         {
             ValidateObject(destination, nameof(destination));
             GaxPreconditions.CheckNotNull(source, nameof(source));
+
+            if (options?.RetentionMode != null || options?.RetainUntilTime != null)
+            {
+                destination.Retention = new Object.RetentionData();
+                if (options.RetentionMode != null)
+                {
+                    destination.Retention.Mode = options.RetentionMode.ToString();
+                }
+                if (options.RetainUntilTime != null)
+                {
+                    destination.Retention.RetainUntilTimeDateTimeOffset = options.RetainUntilTime;
+                }
+            }
```

Comment:
Can't we not do this on the UploadObjectOptions.ModifyMediaUpload method? That's why it's there for.